benchmarks/persistance/s3_persistance.py

```diff
@@ -0,0 +1,89 @@
+"""Classes for persisting and loading experiment results to a S3-Bucket."""
+
+import json
+import os
+
+import boto3
+import boto3.session
+from attr import define, field
+from attrs.validators import instance_of
+from git import Repo
+
+from benchmarks import Benchmark, Result
+
+
+@define
+class S3ExperimentResultPersistence:
+    """Class for persisting experiment results in an S3 bucket."""
+
+    _bucket_name: str = field(validator=instance_of(str), init=False)
+    """The name of the S3 bucket where the results are stored."""
+
+    _object_session = boto3.session.Session()
+    """The boto3 session object. This will load the respective credentials
+    from the environment variables within the container."""
+
+    _branch: str = field(validator=instance_of(str), init=False)
+    """The branch of the Baybe library from which the workflow was started."""
+
+    _workflow_run_id: str = field(validator=instance_of(str), init=False)
+    """The ID of the workflow run."""
+
+    @_bucket_name.default
+    def _default_bucket_name(self) -> str:
+        ENVIRONMENT_NOT_SET = "BAYBE_PERFORMANCE_PERSISTENCE_PATH" not in os.environ
+        if ENVIRONMENT_NOT_SET:
+            raise ValueError(
+                "The environment variable "
+                "BAYBE_PERFORMANCE_PERSISTENCE_PATH is not set. "
+                "A bucket name must be provided."
+            )
+        return os.environ["BAYBE_PERFORMANCE_PERSISTENCE_PATH"]
+
+    @_branch.default
+    def _default_branch(self) -> str:
+        repo = Repo(search_parent_directories=True)
+        current_branch = repo.active_branch.name
+        sanitized_branch = current_branch.replace("/", "-")
+        return sanitized_branch
+
+    @_workflow_run_id.default
+    def _default_workflow_id(self) -> str:
+        if "GITHUB_RUN_ID" not in os.environ:
+            raise ValueError("The environment variable GITHUB_RUN_ID is not set.")
+        return os.environ["GITHUB_RUN_ID"]
+
+    def persist_new_result(self, result: Result, benchmark: Benchmark) -> None:
+        """Store the result of a performance test.
+
+        This method will store the result of a performance test in an S3 bucket.
+        The key of the object will be the experiment ID, the branch, the baybe-version,
```

Comment:
In this case the Database term is meant. S3 is a Key-Value-Store, which means that the Key uniquely identifies the value. You know that from Pythons dict because the same concept is used but more from a hashmap perspective. The object does mean the result and its further information, I thought it might make since since it is stored as a JSON, But as always: Naming suggestions are really welcome 😄

Reply:
If these are well known-terms, then it would already be sufficient to just have `The key (in the database)` or something similar - although this feels kind of redundant. So would also be fine for me to stick with what we have now, feel free to either resolve directly or make a minimal adjutment

Reply:
I've changed it to `S3-Key` which might be more understandable.